public_budget/models/budget_position.py

```diff
@@ -387,3 +387,26 @@ def _compute_assignment_position(self):
             else:
                 assignment_position = rec.get_parent_assignment_position()
             rec.assignment_position_id = assignment_position.id
+
+    @api.multi
```

Comment:
Perfecto nico, tal vez modificaría un poco el código acá, algo así haría:
@api.multi
def action_position_analysis_tree(self):
    self.ensure_one()
    res = {}
    if self.child_ids:
        action = self.env.ref('public_budget.action_position_analysis_tree')
        res = action.read()[0]
        res['domain'] = [('id', 'in', self.child_ids.ids)]
        res['target'] = 'current'
    elif self.preventive_line_ids:
        action = self.env.ref('public_budget.action_budget_position_items')
        res = action.read()[0]
        res['context'] = {
            'search_default_budget_position_id': [self.id],
            'search_default_budget_id': self._context.get(
                'budget_id', False)}
        res['target'] = 'current'
    return res

NOTA: saqué los if action, porque en realidad si no hay action el ref ya devuelve un raise (podes verlo acá https://github.com/odoo/odoo/blob/11.0/odoo/api.py#L791), ese if seguramente antes de que exista el ref y oteniamos el action de otra manera